src/Network/NWConnectionGroup.cs

```diff
@@ -0,0 +1,276 @@
+						context.Handle,
+						null);
+					return;
+				}
+
+				BlockLiteral block_handler = new BlockLiteral ();
+				block_handler.SetupBlockUnsafe (static_SendCompletion, handler);
+				try {
+					nw_connection_group_send_message (GetCheckedHandle (),
+						content == null ? IntPtr.Zero : content.Handle,
+						endpoint == null ? IntPtr.Zero : endpoint.Handle,
+						context.Handle,
+						&block_handler);
+				} finally {
+					block_handler.CleanupBlock ();
+				}
+			}
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		unsafe static extern void nw_connection_group_set_receive_handler (OS_nw_connection_group group, uint maximum_message_size, bool reject_oversized_messages, BlockLiteral *handler);
+
+		delegate void nw_connection_group_receive_handler_t (IntPtr block, IntPtr content, IntPtr context, bool isCompleted);
+		static nw_connection_group_receive_handler_t static_ReceiveHandler = TrampolineReceiveHandler;
+
+		[MonoPInvokeCallback (typeof (nw_connection_group_receive_handler_t))]
+		static void TrampolineReceiveHandler (IntPtr block, IntPtr content, IntPtr context, bool isCompleted)
+		{
+			var del = BlockLiteral.GetTarget<NWConnectionGroupReceiveDelegate> (block);
+			if (del != null) {
+				var nsContent = new DispatchData (content, owns: false);
+				var nsContext = new NWContentContext (context, owns: false);
+				del (nsContent, nsContext, isCompleted);
+				nsContext.Dispose ();
+				nsContext.Dispose ();
+			}
+		}
+
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public void SetReceiveHandler (uint maximumMessageSize, bool rejectOversizedMessages, NWConnectionGroupReceiveDelegate? handler)
+		{
+			unsafe {
+				if (handler == null) {
+					nw_connection_group_set_receive_handler (GetCheckedHandle (), maximumMessageSize, rejectOversizedMessages, null);
+					return;
+				}
+
+				BlockLiteral block_handler = new BlockLiteral ();
+				block_handler.SetupBlockUnsafe (static_ReceiveHandler, handler);
+				try {
+					nw_connection_group_set_receive_handler (GetCheckedHandle (), maximumMessageSize, rejectOversizedMessages, &block_handler);
+				} finally {
+					block_handler.CleanupBlock ();
+				}
+			}
+		}
+
+		[DllImport (Constants.NetworkLibrary)]
+		unsafe static extern void nw_connection_group_set_state_changed_handler (OS_nw_connection_group group, BlockLiteral *handler);
+
+		delegate void nw_connection_group_state_changed_handler_t (IntPtr block, NWConnectionGroupState state , IntPtr error);
+		static nw_connection_group_state_changed_handler_t static_StateChangedHandler = TrampolineStateChangedHandler;
+
+		[MonoPInvokeCallback (typeof (nw_connection_group_state_changed_handler_t))]
+		static void TrampolineStateChangedHandler (IntPtr block, NWConnectionGroupState state, IntPtr error)
+		{
+			var del = BlockLiteral.GetTarget<NWConnectionGroupStateChangedDelegate> (block);
+			if (del != null) {
+				var nwError = (error == IntPtr.Zero) ? null : new NWError (error, owns: false);
```

Comment:
Does adding `using` work here?

```suggestion
				using var nwError = (error == IntPtr.Zero) ? null : new NWError (error, owns: false);
```